spec/features/managing_editor_spec.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+
+RSpec.feature "Managing editor permissions" do
+  scenario do
+    given_there_is_a_published_document
+    when_i_dont_have_the_managing_editor_permission
+    and_i_visit_the_document_page
+    and_i_click_on_withdraw
+    then_i_see_a_message_to_ask_my_managing_editor_to_withdraw_content
+    when_i_do_have_the_managing_editor_permission
+    and_i_visit_the_document_page
+    and_i_click_on_withdraw
+    then_i_can_see_a_public_explanation_form
+
+    given_there_is_a_withdrawn_document
+    when_i_dont_have_the_managing_editor_permission
+    and_i_visit_the_withdrawn_document_page
+    and_i_click_on_change_public_explanation
+    then_i_see_a_message_to_ask_my_managing_editor_to_withdraw_content
+    when_i_do_have_the_managing_editor_permission
+    and_i_visit_the_withdrawn_document_page
+    and_i_click_on_change_public_explanation
+    then_i_can_see_a_public_explanation_form
```

Comment:
This looks like it's two scenarios here. We have an example of this already in: https://github.com/alphagov/content-publisher/blob/25bdfc25a19ad86cee8f0c422a7b9d94445e474a/spec/features/workflow/edition_spec.rb

I also wonder if we need to check that things do work when they have the permission as doesn't that mean we then replicate tests we already have?